modules/nf-core/openms/filefilter/main.nf

```diff
@@ -0,0 +1,51 @@
+process OPENMS_FILEFILTER {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/openms:3.2.0--haddbca4_4' :
+        'biocontainers/openms:3.2.0--haddbca4_4' }"
+
+    input:
+    tuple val(meta), path(file)
+
+    output:
+    tuple val(meta), path("*.{mzML,featureXML,consensusXML}"), emit: filtered
+    path "versions.yml"                                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = task.ext.suffix ?: "${file.getExtension()}"
```

Comment:
It might be better to have the suffix as an input value to make the pipelines more readable

Reply:
I actually like that idea, however would make the module less generic and could cause additional mapping/channel manipulations if you would want to do 

`channel([meta,mzML]) -> filefilter([meta, mzML])`

it would be

`channel([meta,mzML]) -> filefilter(channel([meta,mzML]).map{meta, mzML, -> [meta, mzML, "mzML"]})`

Reply:
I think option 1 is still more readable

Reply:
Maybe if you add the suffix as an input value separate from the first input channel? Or would the suffix be dependent on the samples?

Reply:
True that could be an option! No, not dependent on the sample, only input_suffix == output_suffix

Reply:
Aha okay that makes it even simpler :) You could check the input suffix and make the output suffix the same

Reply:
You mean like this? :)

Reply:
Yeah it looks good as it is now :) You could even remove the input suffix option if the extension has to always be the same as the input

Reply:
Ah, great. Yes that is what I initially had in mind :D perfect

Reply:
I'll remove the validity checks for correct file extension as well, since I realized the tool itself does it already. No need for duplicating this